### PR TITLE
Categorize scripts in `package.json`

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -116,18 +116,18 @@
...
+    "lint:changelog": "node ./scripts/lint-changelog.js",
+    "lint:prettier": "prettier \"**/*.{md,json,yml,html,css}\" --check",
+    "lint:dist": "eslint --no-eslintrc --no-ignore --env=es6,browser \"dist/!(bin-prettier|index|third-party).js\"",
+    "lint:spellcheck": "cspell {bin,scripts,src}/**/*.js {docs,website/blog,changelog_unreleased}/**/*.md",
+    "lint:deps": "node ./scripts/check-deps.js",
     "build": "node --max-old-space-size=3072 ./scripts/build/build.js",
-    "build-docs": "node ./scripts/build-docs.js",
-    "check-deps": "node ./scripts/check-deps.js",
-    "spellcheck": "cspell {bin,scripts,src}/**/*.js {docs,website/blog,changelog_unreleased}/**/*.md"
+    "build-docs": "node ./scripts/build-docs.js"
```
Comment (fisker) [inline package.json]: I guess this one is using somewhere on netlify. Should not be able to rename.

**Comment by fisker** [comment]
@thorn0 How about `lint:type-check` , since `typecheck` is not a word.

**Comment by thorn0** [comment]
All three spellings are used: type check, [type-check](https://en.wiktionary.org/wiki/type-check), [typecheck](https://en.wiktionary.org/wiki/typecheck).

**Comment by fisker** [comment]
Good to know.

**Pull request merged**